CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+- Arrays
+  - Array of a single type (`Array<number>`)
+  - Array of a union type (`Array<string | number>`)
+  - Array of literal types (`Array<"cat" | "dog">`)
+  - Union of similar arrays (`Array<1 | 2> | Array<2 | 3>`)
+  - Nested arrays (`Array<Array<number>>`)
+- Tuples
+  - Tuple of a single type (`[number, number, number]`)
+  - Tuple of different types (`[number, string, boolean]`)
+  - Tuple with optional type (`[number, string?]`)
+  - Tuple with spreads (`[...Array<number>, number, number]`)
+  - Tuple with optional type and spread (`[number?, ...Array<string>]`)
+  - Tuple with union values (`[number, string | undefined, boolean]`)
+  - Tuple with literal values (`[number, "cat" | "dog", true]`)
+- Records
+  - Record with string keys (`Record<string, number>`)
+  - Record with number keys (`Record<number, string>`)
+  - Record with union keys (`Record<string | number, unknown>`)
+  - Record with union values (`Record<string, string | number>`)
+  - Record with undefined values (`Record<string, string | undefined>`)
+  - Record with literal keys (`Record<"cat" | "dog", number>`)
+  - Record with literal values (`Record<string, 1 | 2>`)
+  - Record with template keys (``Record<`id_${number}`, string>``)
+  - Union of records (`Record<string, unknown> | Record<number, unknown>`)
+- Objects
+  - Object with named keys (`{ a: number }`)
+  - Object with union values (`{ a: string | number }`)
+  - Object with literal union values (`{ a: "cat" | 1 }`)
+  - Object with optional keys (`{ a?: number }`)
+  - Object with symbol keys (`{ [Symbol("a")]: number }`)
+  - Union of objects (`{ a: 1, b: 2 } | { b: 2, c: 3 }`)
+  - Nested objects (`{ a: { b: { c: 1 } } }`)
+- `readonly` versions of the above
+- `null` and `undefined`
+
+### Design philosophy
+
+**No type annotations.** Functions shouldn't require type annotations to have good types. This gives the best developer experience. The exception is `piped`.
+
+**Have good output types.** One reason Remeda provide "one-liner" functions is if we can have better output types than the defaults. While `data.filter(item => typeof item === 'number')` works, `filter(data, isNumber)` gives a better output type. This is why we have type guards, `hasAtLeast`, `prop`, etc.
+
+**Readability counts.** The other reason Remeda provides one-liner functions is if they make things more readable. Compare `map((item) => item + 3)` to `map(add(3))`, where we don't need to come up with a name for `item`. This is why we have `constant`, `doNothing`, `identity`, etc.
+
+**Prefer composition.** We don't provide simple compositions of functions. Having a smaller surface area makes it easier for users to remember Remeda's functions, and easier for us to maintain. We don't have `reject` because it's the same as `filter(isNot)`. An exception is when the composition would be less readable; this is why we have `sum`, even if it's the same as `sumBy(identity)`.
```

Comment:
`sum` isn't the best example because I still think it's redundant (but I agree that right now, the alternative isn't great). Maybe use `compact` as a second example, or `zipObj`